### ISSUE TITLE
J2534 (passthrucan) connection does not stay connected #208

### ISSUE THREAD

**Issue by collin80** [collin80/SavvyCAN#563]
### Discussed in https://github.com/collin80/SavvyCAN/discussions/560

<div type='discussions-op-text'>

<sup>Originally posted by **Tmod700** January  2, 2023</sup>
I got x64 Build 208 to show the port selections when choosing a passthru connection but it connects and then disconnects itself just a few seconds later. I am using Tactrix Openport 2.0 and Win10x64 and downloaded the 208 CIBuild.
I tried 207 compiled for 32bit and it connects and stays connected fine.
Any ideas as to why it auto disconnects?
![Ashampoo_Snap_2023 01 01_12h08m01s_001_](https://user-images.githubusercontent.com/79778536/210200526-3de0ad7d-fe27-478d-90aa-5ace5d59a737.png)
</div>

**Comment by Bill400** [comment]
Using Visual Studio 2022 to examine the V208 source files.
Connections module\Canbus.cpp appears to contain our default bus speed of 250Kbts. Statements that follow appear to set the operating parameters. Not yet the C++ expert, I wonder about the last two statements, which are identical except for the const keyword which specifies that the object or variable isn't modifiable. Is the final statement without the const keyword an error? Should it have been deleted?
Further, a work around (for 500Kbts) might be to change the default 250000 to 500000?

**Comment by amiralnar** [comment]
> Hi guys I have OpenPort 2.0 clone from China and I was able to compile 32bit version with hardcoded 500kb/s speed for "passthrough" connection and it worked like a charm while i send debug frames from my desktop PC. But the same compiled binary running on my laptop behaves exactly like Collin said - connection drops in a few seconds after it was "Connected". Both devices are running Windows 10 x64, both have the same 32bit OpenPort driver. I feel the main difference is that laptop have no Qt SDK installed and binary uses some included lib. Anyway i'll try to investigate that deeply and who knows maybe i'll find a solution for that
> 
> EDIT: Nevermind. I just reinstalled OpenPort driver (i believe exactly the same driver was installed, strange thing) and now my adapter remains connected. Now it fully covers my needs, Thank you for a great application!

How did you hardcoded speed exactly? @macbaze

**Comment by dmg210** [comment]
@macbaze Any chance you could share the complied 32bit version with the 500kb/s hardcoded? Thanks.

**Comment by amiralnar** [comment]
> @macbaze Any chance you could share the complied 32bit version with the 500kb/s hardcoded? Thanks.

Try this one: https://github.com/amiralnar/SavvyCAN/releases/download/continuous/SavvyCAN-Windows_x32_CIBuild.zip

**Comment by dmg210** [comment]
Legend! Thanks. That's working a treat with the Tactrix OpenPort 2.0 👍

**Comment by mrdka69** [comment]
Did you guys connect pin 6 and 14 of the openport 2.0 to the can H and can L of the given device? Im trying to do the same byt unfortunately dont get anything in savvycan. Ive downloaded the savycan from @amiralnar

**Comment by dmg210** [comment]
Yeah I just used the regular obd pinout on the openport. Pin 6 - can high, pin 14 - can low.

**Comment by rovis-hub** [comment]
В новій версії взагалі не бачить прилади по passthrucan. Збірка розробки QT6 [https://github.com/collin80/SavvyCAN/releases/tag/continuous-QT6](url)
